cli/azd/pkg/tools/azcli/subscriptions.go

```diff
@@ -164,14 +165,15 @@ func (s *SubscriptionsService) ListTenants(ctx context.Context) ([]armsubscripti
 		}
 
 		for _, tenant := range page.TenantListResult.Value {
-			if tenant != nil {
+			if tenant != nil && tenant.TenantID != nil {
 				tenants = append(tenants, *tenant)
 			}
 		}
 	}
 
 	sort.Slice(tenants, func(i, j int) bool {
-		return *tenants[i].DisplayName < *tenants[j].DisplayName
+		return convert.ToValueWithDefault(tenants[i].DisplayName, "") <
```

Comment:
Why do we need to `sort` tenants, anyway?

This might be a side effect of using `ListSubscriptions` as the structure for `ListTenants` ?   

Since we are never showing the list of tenants to users, we could get rid of this `sort`

Reply:
I believe the original idea is to make any type of listing as deterministic as possible. As you said, there is no consumption today and thus makes less sense. sorting should really be done at the UI layer. That being said, I'm going to defer this outside of this fix, since changing that is a design change.